commit bfc6eaf4dc7a486e1300ef27f8dc380ffacebec0 (refs/changes/07/3507/1)
Author: Tim Kordas <tim.kordas@joyent.com>
Date:   2018-02-27T15:46:36-08:00 (1 year, 7 months ago)
    
    OS-6328 mdb should not load the same dmod more than once

diff --git a/usr/src/cmd/mdb/common/mdb/mdb_module.c b/usr/src/cmd/mdb/common/mdb/mdb_module.c
index 286f9d0074..5c69ad9bfb 100644
--- a/usr/src/cmd/mdb/common/mdb/mdb_module.c
+++ b/usr/src/cmd/mdb/common/mdb/mdb_module.c
@@ -124,8 +124,13 @@ mdb_module_create(const char *name, const char *fname, int mode,
 
 	if (!(mode & MDB_MOD_BUILTIN)) {
 		mdb_dprintf(MDB_DBG_MODULE, "dlopen %s %x\n", fname, dlmode);
+		mod->mod_hdl = dlmopen(LM_ID_BASE, fname, RTLD_NOLOAD);
+		if (mod->mod_hdl != NULL) {
+			/* We don't re-init the same module twice. */
+			warn("%s: already loaded\n", name);
+			goto err;
+		}
 		mod->mod_hdl = dlmopen(LM_ID_BASE, fname, RTLD_NOW | dlmode);
-
 		if (mod->mod_hdl == NULL) {
 			warn("%s\n", dlerror());
 			goto err;
